tests/audioworklet/audioworklet.cpp

```diff
@@ -0,0 +1,190 @@
+  const float MAXAMP = 0.2f; // 20% so it's not too loud
+
+  float* out = outputBuffer;
+  while(numSamples > 0) {
+    // Amplitude at current position
+    float a = sinf(wavePos) * MAXAMP; 
+  
+    // Advance position, keep it in [0, 2*PI) range to avoid running out of float precision
+    wavePos += invSampleRate * FREQ * PI2;
+    if(wavePos > PI2) {
+      wavePos -= PI2;
+    }
+    
+    // Set both left and right samples to the same value
+    out[0] = a;
+    out[1] = a;
+    out += 2;
+
+    numSamples -= 1;
+  }
+
+  return outputBuffer;
+}
+
+// Initializes the audio context and the pthread it it's AudioWorkletGlobalScope
+EM_JS(uint32_t, initAudioContext, (pthread_t* pthreadPtr), {
+  // Create the context
+  Module.audioCtx = new AudioContext();
+
+  // To make this example usable we setup a resume on user interaction as browsers
+  // all require the user to interact with the page before letting audio play
+  if (window && window.addEventListener) {
+    var opts = { capture: true, passive : true };
+    
+    var resume = function () {
+      Module.audioCtx.resume();
+      window.removeEventListener("touchstart", resume, opts);
+      window.removeEventListener("mousedown", resume, opts);
+      window.removeEventListener("keydown", resume, opts);
+    };
+    
+    window.addEventListener("touchstart", resume, opts);
+    window.addEventListener("mousedown", resume, opts);
+    window.addEventListener("keydown", resume, opts);
+  }
+
+  // Initialize the pthread shared by all AudioWorkletNodes in this context
+  PThread.initAudioWorkletPThread(Module.audioCtx, pthreadPtr).then(function() {
+    out("PThread context initialized!")
+  }, function(err) {
+    out("PThread context initialization failed: " + [err, err.stack]);
+  });
+
+  return Module.audioCtx.sampleRate;
+});
+
+// Creates an AudioWorkletNode and connects it to the output once it's created
+EM_JS(void, createAudioWorklet, (), {
+  PThread.createAudioWorklet(
+    Module.audioCtx,
+    'native-passthrough-processor', 
+    {
+      numberOfInputs: 0,
+      numberOfOutputs : 1,
+      outputChannelCount : [2]
+    }
+  ).then(function(workletNode) {
+    // Connect the worklet to the audio context output
+    out("Audio worklet created!");
+    workletNode.connect(Module.audioCtx.destination);
+  }, function(err) {
+    out("Audio worklet creation failed: " + [err, err.stack]);
+  });
+});
+
+
+int main()
+{
+  // Section 1 - test-only code, ignore 
+  setupUnloadHandler();
+  emscripten_request_animation_frame_loop(mainLoop, nullptr);
+  testMutex.lock(); // Lock the mutex so that the worklet thread can wait on it (see comment in 'signalTestSuccess')
+
+  // Section 2 - audio worklet example
+  sampleRate = initAudioContext(&workletThreadId);
+  invSampleRate = 1.0f / sampleRate;
+  printf("Initialized audio context. Sample rate: %d. PThread init pending.\n", sampleRate);
+  createAudioWorklet();
+  printf("Creating audio worklet.\n");
```

Comment:
Shouldn't this print come before the call to `createAudioWorklet` (or be phrased in the past tense).